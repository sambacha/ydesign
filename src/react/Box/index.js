import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  compose,
  color,
  space,
  layout,
  position,
  flexbox,
  border,
  typography,
  shadow,
} from 'styled-system';
import propTypes from '@styled-system/prop-types';
import theme from '../theme';

const styleProps = compose(
  color,
  space,
  layout,
  position,
  flexbox,
  border,
  typography,
  shadow
);

const StyledBox = styled('div')`
  box-sizing: border-box;
  ${styleProps}
`;

StyledBox.defaultProps = {
  theme,
};

const Box = React.forwardRef((props, ref) => (
  <StyledBox ref={ref} {...props} />
));

Box.propTypes = {
  ...propTypes.styleProps,
  theme: PropTypes.object,
};

Box.displayName = 'Box';

export { StyledBox };

export default Box;
