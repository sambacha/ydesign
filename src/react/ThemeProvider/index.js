import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
