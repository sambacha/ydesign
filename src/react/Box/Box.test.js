import Box from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Box component sanity', () => {
  it('has name', () => {
    expect(Box.displayName).toBe('Box');
  });

  it('matches default snapshot', () => {
    const component = render(<Box />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Box />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
