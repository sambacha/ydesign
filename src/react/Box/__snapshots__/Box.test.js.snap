// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
}

<div
  class="c0"
/>
`;

exports[`Box component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
}

<div
  class="c0"
/>
`;
